elastic/kibana
# [Dashboard] [Control] Remove support for scripted fields for options list

Closes https://github.com/elastic/kibana/issues/144631

## Summary

As described in the attached issue, the options list control was broken for scripted fields - more specifically, when creating an options list control on a scripted field, the dropdown would simply always show no results:

<p align="center">
  <img src="https://user-images.githubusercontent.com/8698078/200067075-0478bff9-9b98-45fc-a584-5005b09923df.png">
</p>

Rather than dedicating significant engineering effort towards adding this functionality for a feature that was [deprecated back in 7.13](https://www.elastic.co/guide/en/kibana/current/managing-data-views.html#scripted-fields), we have decided to simply make scripted fields incompatible with options list controls.

### Videos
#### Before

https://user-images.githubusercontent.com/8698078/200068300-05bb64ca-c4d3-49ed-b4e4-7b6269d8f020.mov


#### After

https://user-images.githubusercontent.com/8698078/200068313-9414d29f-aaf1-4f09-869e-7404c9ec7150.mov


#### Notes
- Because scripted fields work fine with the range slider control, these have not been impacted. 
- Runtime fields are still supported for the options list:
     
<p align="center">
  <img src="https://user-images.githubusercontent.com/8698078/200069622-338f2ca1-76bf-4676-895c-6aadd8dc7f0c.png">
</p>

###  How to Test

#### Testing Scripted Fields
1. Here is a sampling of the scripted fields of different types I created with the Logs sample data:
     - **Boolean:** `successful` -> `doc['response.keyword'].value == "200"`
     - **String:** `referer_domain` -> `doc['referer'].value.splitOnToken('/')[2]`
     - **Number:** `year` -> `doc['utc_time'].value.toString().splitOnToken('-')[0]`
2. Ensure that any scripted fields whose type would typically be supported by the options list control (boolean, string, etc.) do not show up in the field picker when creating/editing a control
3. Ensure that scripted number fields still show up, since the range slider should still support scripted fields

#### Testing Runtime Fields
1. Add a string runtime field to the Logs sample data as follows: <br/><br/>
    ```
    PUT kibana_sample_data_logs/_mapping
    {
      "dynamic": "runtime",
      "runtime": {
        "runtime__referer_domain": {
          "type": "keyword",
          "script": {
            "source": "emit(doc['referer'].value.splitOnToken('/')[2])"
          }
        }
      }
    }
    ```
2. Hard refresh Kibana
3. Ensure that the runtime field still show up in the field picker when creating/editing a control

#### Flaky Test Runner

<a href="https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1517)"><img src="https://user-images.githubusercontent.com/8698078/200083488-ed8a9dc1-de58-4a42-95b8-a3b20de033c3.png"/></a>



### Checklist



### For maintainers
